tests/integration/test_deploy_aws.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+CHARM_NAME = METADATA["name"]
+
+
+class TestDeployAWS:
+    @pytest.mark.abort_on_fail
+    async def test_001_build_and_deploy(self, ops_test: OpsTest):
+        """Build and deploy the charm.
+
+        Assert on the unit status.
+        """
+        await ops_test.model.deploy(CHARM_NAME, channel="latest/edge", trust=True)
```

Comment:
Because there's one line of code different between `test_deploy_aws.py` and `test_charm.py`, this jumps out at me as a case where there's probably a better way.  Some ideas that come to mind:
* pass arguments into the test to change the behaviour.  Perhaps something like obs where they optionally pass the path to an already built charm?  Could do something like that with an optional path to a charm that needs pulling?  
* use marks or some other control flow built into pytest.  That at least avoids multiple test files.  You can have a default path for typical use, and a "please use aws instead" path for the special case
* either for marks or for separate files, you can still share the bulk of the test code.  Something like:

```python
def test_on_aws():
  some_configurations = aws_specific_configurations
  build_and_deploy_helper(some_configurations)

def test_normally():
  some_configurations = normal_configurations
  build_and_deploy_helper(some_configurations)

def build_and_deploy_helper(...):
  # this has all the ops_test.model.deploy, wait, etc stuff.  That way it is shared between both
```

Reply:
+1 to remove duplication via method extraction

however, not clear the idea of files. I think there is a value in keeping different files.
for example if you would like to create tests that run on AWS only, then having separate file for AWS will be cleaner

Reply:
Yeah @ca-scribner  we had this discussion in bundle tests release tests where we decided marks are not a good way so we should rather use separate files .... I think separate files are good way here as there might be more testing connected with aws specifically.

Reply:
I am in for separate files. Makes it a bit more readable and independent. Also, if we grow test suites for AWS only or Azure only with specific tests, I wouldn't want to have them in regular integration tests. It will make regular integration test files for development too big and unreadable.

Reply:
there's two orthogonal conversations here that can be decided separately:
* should we use helpers like demo'd above
* should we use marks or separate files

Sounds like we agree that we should use method extraction so that `test_on_aws()` and `test_normally()` differ only on the configuration they set, and then they share the helper for actually doing the building and deploying.  Agreed?

Re marks vs files: we go with separate files

Reply:
I would disagree with this *test_on_aws() and test_normally() differ only on the configuration they set*.
What if my AWS tests will utilize AWS specific feature in the integration test, eg. use "awscli" or connect to AWS service?
Ideally, the interfaces we use should be abstract enough for us to not care about what cloud is out there, but reality is different as we know.

Reply:
+1 for shaing helpers for building/deploying
+1 for anything that can be generalized to be move to `test_charm.py`